fix(tui): don't disable vsplit scrolling in all xterm-like terminals

bfredl opened the pull request:
Current "patch" asserts that EVERY terminal claiming to be an xterm has broken vertical split scrolling which is far too an aggressive workaround. This should be disabled only for specific terminals known to lie about being "xterm" alike when they are not. Users who use properly implemented terminals should enjoy _hardware accelerated scrolling_ even with vertical split windows.

The latest time this was touched #16239, it was due to concerns about VTE #16238 but this has been fixed in later versions of VTE.

But if there are other terminals which needs to be shamed besides `hterm` these can be added as specific disables.

bfredl committed:
fix(tui): feature detect vsplit scrolling in xterm-like terminals

Current "patch" asserts that EVERY terminal claiming to be an xterm has
broken vertical split scrolling which is far too an aggressive
workaround.

Instead, as left-right margins for scrolling regions use a
private DEC mode, we can use feature detection to use them in supported
terminals only. This way, users who use properly implemented terminals
can enjoy hardware accelerated scrolling even with vertical split
windows.

bfredl commented:
This is also DEC mode 69 DECLRMM which in principle should be detectable (nice)

clason commented on src/nvim/tui/tui.c:
```diff
@@ -229,16 +230,20 @@ void tui_handle_term_mode(TUIData *tui, TermMode mode, TermModeState state)
-  case kTermModePermanentlySet:
   case kTermModePermanentlyReset:
+    // TODO(bfredl): This is really ILOG but we want it in all builds.
+    // add to show_verbose_terminfo() without being too racy ????
```
Off-topic: should we add a terminal feature section to `:checkhealth`? (@gpanders)

bfredl commented:
updated version: use _hardware accelerated vsplits_ only when terminal reports support via DECRQM.

![image](https://github.com/user-attachments/assets/18a1f871-d8fe-460d-98bb-d1454edac1de)

bfredl committed:
refactor(tui): remove obsolete can_set_left_right_margin duplicate

According to terminfo(5), these are obsolete:

Except for very old terminal descriptions, e.g., those developed
for SVr4, the scheme just described should be considered obsolete.
An improved set of capabilities was added late in the SVr4
releases (smglr and smgtb), which explicitly use two parameters
for setting the left/right or top/bottom margins.

bfredl commented:
deleted the duplicate `can_set_left_right_margin` implementation. it sounds like the separate left margin and right margin sequences are only relevant for setting the marginal of old printers(!). anything vaguely screen terminal like will either support the `\x1b[%i%p1%d;%p2%ds` form or not at all
